compiler/src/dotty/tools/dotc/typer/RefChecks.scala

```diff
@@ -1166,7 +1166,9 @@ object RefChecks {
       def targetOfHiddenExtension: Symbol =
         val target =
           val target0 = explicitInfo.firstParamTypes.head // required for extension method, the putative receiver
-          target0.dealiasKeepOpaques.typeSymbol.info
+          target0.dealiasKeepOpaques.typeSymbol.info match
```

Comment:
This is kind of weird since dealias does skip aliasing bounds. It would be good to get an explanation what exactly goes wrong here. What's the type of `target0.dealiasKeepOpaques` in the failing test? What is its info? Is there an opaque type involved?

Reply:
The ~correct~ working idiom is
```
explicitInfo.firstParamTypes.head.typeSymbol.typeRef.dealiasKeepOpaques
```
but requires reworking https://github.com/scala/scala3/pull/22502/files
or `dealias` should handle `TypeAlias` directly.